### DISCUSSION TITLE
Linux: How to link target libraries such as PhysXExtensions_static_64, PhysX_static_64

### DISCUSSION THREAD

**Discussion by zoctipus** [NVIDIA-Omniverse/PhysX#235] [Q&A]
Hello, I am a C++ Beginner, I want to make the SnippetHelloWorld.cpp work. I want to ask how to link target libraries such as PhysXExtensions_static_64, PhysX_static_64 as showned in `https://nvidia-omniverse.github.io/PhysX/physx/5.3.1/docs/BuildingWithPhysX.html`

What I have done sofar

 1. git clone git@github.com:NVIDIA-Omniverse/PhysX.git
 2. cd physx
 3. ./generate_project.sh  ->input 1 for linux
 4.  in root folder create CMakeLists.txt as below
 5.  $cmake .  then   $make

```
cmake_minimum_required (VERSION 3.1)
project (PhysX_Standalone_Snippet)

option(PHYSX_BUILD_TYPE "The build type of PhysX, i.e., one of {debug, checked, profile, release}" "checked")

include_directories(PhysX/include)

if(CMAKE_BUILD_TYPE AND CMAKE_BUILD_TYPE STREQUAL "Debug")
    message("Building snippet in debug configuration")
    add_compile_definitions(_DEBUG)
    link_directories("PhysX/bin/linux.clang/debug") # This is the path where PhysX libraries are installed
else()
    message("Building snippet in release configuration with PhysX ${PHYSX_BUILD_TYPE} configuration")
    add_compile_definitions(NDEBUG)
    link_directories("PhysX/bin/linux.clang/${PHYSX_BUILD_TYPE}") # This is the path where PhysX libraries are installed
endif()

add_executable(Snippet
    main.cpp
)

target_link_libraries(Snippet
    PhysXExtensions_static_64
    PhysX_static_64
    PhysXPvdSDK_static_64
    PhysXCommon_static_64
    PhysXFoundation_static_64

    pthread
    dl
)
```

however what I ended up is 

```
(base) octipus@octipus:~/Projects/PhysX$ make
[ 50%] Linking CXX executable Snippet
/usr/bin/ld: cannot find -lPhysXExtensions_static_64: No such file or directory
/usr/bin/ld: cannot find -lPhysX_static_64: No such file or directory
/usr/bin/ld: cannot find -lPhysXPvdSDK_static_64: No such file or directory
/usr/bin/ld: cannot find -lPhysXCommon_static_64: No such file or directory
/usr/bin/ld: cannot find -lPhysXFoundation_static_64: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/Snippet.dir/build.make:97: Snippet] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/Snippet.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

I searched hours and still don't know the solution. Big thanks for helping me!

**Comment by preist-nvidia** [comment]
Hi @zoctipus - Cmake setups can be tricky, yes. We don't have support bandwidth to help with this specific issue, unfortunately.

Maybe you can just use the vcpkg described here? https://omni-101.github.io/omniverse/2023/07/07/omniverse-physx-vcpkg-port/

**Comment by hvandamm** [comment]
I pressed 0 because i used the clang compiler but the difference with 1 should not be big.

Go to PhysX/physx/compiler/linux-checked. (if you pressed 1 i guess it would be PhysX/physx/compiler/linux-aarch64-checked) You see a Makefile there. Type "make" and wait for the compilation to end.
After the compilation ended you should see the static libraries (ending with .a) in physx/bin/linux.clang/checked directory (for you maybe it is physx/bin/linux.aarch64/checked or something). you now can use this path to tell Cmake where the static libraries are.

so if the static libraries are in physx/bin/linux.clang/checked then change the line "link_directories("PhysX/bin/linux.clang/${PHYSX_BUILD_TYPE}")" in Cmake to "link_directories("physx/bin/linux.clang/checked")"

keep in mind that these paths are relative paths, relative to where you type make when in the root folder.

Hope this works.
